src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java

```diff
@@ -810,28 +810,30 @@ public String getDlsQuery(User user) {
         public boolean hasDlsQuery() {
             return dlsQuery != null && !dlsQuery.isEmpty();
         }
-        
+
         public Set<String> getFls() {
             return Collections.unmodifiableSet(fls);
         }
 
         public boolean hasFlsFields() {
             return fls != null && !fls.isEmpty();
         }
-        
+
         public Set<String> getMaskedFields() {
             return Collections.unmodifiableSet(maskedFields);
         }
 
         public boolean hasMaskedFields() {
             return maskedFields != null && !maskedFields.isEmpty();
         }
-        
+
         public WildcardMatcher getPerms() {
             return WildcardMatcher.from(perms);
         }
 
-        
+        public WildcardMatcher getNonWildCardPerms() {
```

Comment:
Can either remove this here or remove the second check for * above. (See previous comment).